tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas2_gemv_kk.hpp

```diff
@@ -0,0 +1,228 @@
+
+#ifndef LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_GEMV_HPP_
+#define LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_GEMV_HPP_
+
+namespace KokkosKernelsSTD {
+
+namespace gemvimpl{
+template<class Accessor>
+double get_scaling_factor(Accessor) { return 1.0; }
+
+template<class Accessor, class S>
+auto get_scaling_factor(std::experimental::linalg::accessor_scaled<Accessor,S> a) {
+  return a.scale_factor();
+}
+} //end anon namespace
+
+//
+// overwriting gemv: y = Ax
+//
+// for now, specialize for default_accessor
+// https://github.com/kokkos/stdBLAS/issues/122
+//
+template<class ExeSpace,
+         class ElementType_A,
+         std::experimental::extents<>::size_type numRows_A,
+         std::experimental::extents<>::size_type numCols_A,
+         class Layout_A,
+         class ElementType_x,
+         std::experimental::extents<>::size_type ext_x,
+         class Layout_x,
+         class ElementType_y,
+         std::experimental::extents<>::size_type ext_y,
+         class Layout_y>
+void matrix_vector_product(kokkos_exec<ExeSpace> /*kexe*/,
+			   std::experimental::mdspan<
+			     ElementType_A,
+			     std::experimental::extents<numRows_A, numCols_A>,
+			     Layout_A,
+			     std::experimental::default_accessor<ElementType_A>
+			   > A,
+			   std::experimental::mdspan<
+			     ElementType_x,
+			     std::experimental::extents<ext_x>,
+			     Layout_x,
+			     std::experimental::default_accessor<ElementType_x>
+			   > x,
+			   std::experimental::mdspan<
+			     ElementType_y,
+			     std::experimental::extents<ext_y>,
+			     Layout_y,
+			     std::experimental::default_accessor<ElementType_y>
+			   > y)
+{
+
+  // constraints
```

Comment:
"Constraints" are enforced through SFINAE, concepts, etc.  "Mandates" are enforced via `static_assert`.  (I'll take this for now, if the comment says something like "FIXME: enforce constraints as such, not as mandates.")

Reply:
Thanks for pointing it out, I can work on changing it. 
Btw, off the top of my head I think there are a few places I saw inside the reference implementation where the same inconsistency is present, namely static asserts for constraints, but i can double check.

Reply:
The reference implementation needs more work, for sure : - )  I'm just hoping that new code can Do The Right Thing, so we have less old code to fix in the future.  Thanks!